### PR TITLE
test: Code Coverage: My View Report Component #2

### PR DESCRIPTION
**Pull request by jayfyle**
### Clickup: https://app.clickup.com/t/85zt055va

# Coverage at 99%

my_view_report_99

### CODE REVIEW

File: src/app/fyle/my-view-report/my-view-report.page.spec.ts [hunk 1]
```diff
@@ -268,42 +293,310 @@ describe('MyViewReportPage', () => {
...
+    it('should return undefined if settings not provided', () => {
+      const result = component.getSimplifyReportSettings(null);
+
+      expect(result).toBeUndefined();
+    });
+  });
+
+  describe('ionViewWillEnter():', () => {
+    it('should load report and report status', fakeAsync(() => {
+      const erpt = cloneDeep({ ...expectedAllReports[0], rp_state: 'APPROVER_INQUIRY' });
```
Comment (Dimple16) [inline src/app/fyle/my-view-report/my-view-report.page.spec.ts:326]: are we not mocking `cloneDeep()`?

File: src/app/fyle/my-view-report/my-view-report.page.spec.ts [hunk 2]
```diff
@@ -268,42 +293,310 @@ describe('MyViewReportPage', () => {
...
+    component.updateReportName('My Testing Report');
+    expect(reportService.updateReportDetails).toHaveBeenCalledOnceWith(cloneDeep(reportParam));
     expect(component.loadReportDetails$.next).toHaveBeenCalledTimes(1);
   });
 
-  it('editReportName(): shoudl edit report name', fakeAsync(() => {
-    component.erpt$ = of(expectedAllReports[0]);
-    fixture.detectChanges();
+  describe('editReportName(): ', () => {
+    it('shoudl edit report name', fakeAsync(() => {
```
Comment (Dimple16) [inline src/app/fyle/my-view-report/my-view-report.page.spec.ts:548]: `should`

File: src/app/fyle/my-view-report/my-view-report.page.spec.ts [hunk 3]
```diff
@@ -268,42 +293,310 @@ describe('MyViewReportPage', () => {
...
+      editReportNamePopoverSpy.onWillDismiss.and.resolveTo({ data: { reportName: 'new name' } });
 
-    popoverController.create.and.returnValue(Promise.resolve(editReportNamePopoverSpy));
+      popoverController.create.and.returnValue(Promise.resolve(editReportNamePopoverSpy));
 
-    component.editReportName();
-    tick(5000);
+      const editReportButton = getElementBySelector(fixture, '.view-reports--card-header__icon') as HTMLElement;
+      click(editReportButton);
+      tick(5000);
```
Comment (Dimple16) [inline src/app/fyle/my-view-report/my-view-report.page.spec.ts:562]: do we need`5000`?

File: src/app/fyle/my-view-report/my-view-report.page.spec.ts [hunk 4]
```diff
@@ -671,44 +1038,99 @@ describe('MyViewReportPage', () => {
...
+      ]);
+      addExpensesToReportModalSpy.onWillDismiss.and.resolveTo({
+        data: {
+          selectedTxnIds: ['txfCdl3TEZ7K', 'txWphhAUZbq7'],
+        },
+      });
 
-    component.showAddExpensesToReportModal();
-    tick(5000);
+      const properties = {
```
Comment (Dimple16) [inline src/app/fyle/my-view-report/my-view-report.page.spec.ts:1058]: move this to mock model properties file

**Review by Dimple16 — CHANGES_REQUESTED** [review]
added comments

File: src/app/fyle/my-view-report/my-view-report.page.spec.ts (hunk 1 above)
Comment (jayfyle) [reply src/app/fyle/my-view-report/my-view-report.page.spec.ts:326]: No, no this is to ensure the properties don't get modified and fail some other test

**Review by Dimple16 — APPROVED** [review]

**Review by yksolanki9 — APPROVED** [review]

**Pull request merged**